### DISCUSSION TITLE
Why should we write types for translation keys?

### DISCUSSION THREAD

**Discussion by Divinely5782** [majesticooss/astro-starter#30] [General]
Hi! I am new to TS and have just started to learn it and i have a question. Why do we have write types for translation keys? IMHO it is obvious that all translation in such simple websites are strings?

Thanks for answer!

`declare interface Locales {
	home_description: string;
	about_description: string;
}`

**Comment by zanhk** [comment]
Hi, This can be defined alternatively like this I think

```
type Locales = Record<'home_description' | 'about_description' | 'home' | 'about' | 'blog', string>;
```

otherwise

```
interface Locales {
	[key: string]: string;
}
```


but in the second case you lose type safesty on the locales

**Comment by zanhk** [comment]
either way I suggest to use a different way for localization as this one is obslolete.

Follow

https://docs.astro.build/en/guides/internationalization/

---

I will release a template in the near future which will use the integrated way.
